src/main/scala/org/constellation/infrastructure/endpoints/SnapshotEndpoints.scala

```diff
@@ -173,6 +174,21 @@ class SnapshotEndpoints[F[_]](implicit F: Concurrent[F]) extends Http4sDsl[F] {
         .map(_.asJson)
         .flatMap(Ok(_))
   }
+
+  private def getTotalSupply(snapshotService: SnapshotService[F]): HttpRoutes[F] = HttpRoutes.of[F] {
+    case GET -> Root / "total-supply" =>
+      snapshotService
+        .getTotalSupply()
+        .map {
+          case (height, totalSupply) =>
+            Map(
```

Comment:
Are you confident a Map with correct key names is good enough? As opposed to a case class?

Reply:
it will be parsed in the same way. do you want me to create an additional case class for that?

Reply:
I would create a case class to avoid confusion on our codebase side.